internal/action/delete.go

```diff
@@ -104,7 +107,11 @@ func (s *Action) deleteKeyFromYAML(ctx context.Context, name, key string) error
 	sec.Del(key)
 
 	if err := s.Store.Set(ctxutil.WithCommitMessage(ctx, "Updated Key"), name, sec); err != nil {
-		return exit.Error(exit.IO, err, "Can not delete key %q from %q: %s", key, name, err)
+		if errors.Is(err, store.ErrMeaninglessWrite) {
```

Comment:
A more idiomatic approach would be to handle the returned error first and then the warning, like this:

```
if !errors.Is(err, store.ErrMeaninglessWrite) {
  return ...
}
out.Warningf(...)
```

Both are in the error handling branch of Store.Set anyway and any error but MeaninglessWrite should be returned.

This comment applies to this place and any other instance of this pattern.